external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -576,6 +579,8 @@ public void ack(Object messageId) {
                     + "came from a topic-partition that this consumer group instance is no longer tracking "
                     + "due to rebalance/partition reassignment. No action taken.", msgId);
             } else {
+                //a null tuple should be added to the ack list since by definition is a direct ack
```

Comment:
We should probably switch the order of the emitted contains and msgId isEmitted checks. It is not possible that isEmitted is false while the tuple is in the emitted list.